### PR TITLE
[Instrumentation.AWSLambda] Updated AWSLambda ActivitySource Name

### LINKED ISSUE #484: [Instrumentation] Standardize ActivitySource names for instrumentation projects
Use the assembly name as the `ActivitySourceName` in the instrumentation projects:

For e.g.: OpenTelemetry.Instrumentation.StackExchangeRedis

Currently, the following projects do not follow this convention:

1. StackExchangeRedis
2. EntityFrameworkCore
3. AWS
4. AWS.Lambda

### PR DESCRIPTION
**Pull request by samimusallam**
Fixes #484 
## Changes
Use the assembly name as the ActivitySource name

Please provide a brief description of the changes here.

For significant contributions please make sure you have completed the following items:

* [x] Appropriate `CHANGELOG.md` updated for non-trivial changes
* [ ] Design discussion issue #

### CODE REVIEW

File: src/OpenTelemetry.Contrib.Instrumentation.AWSLambda/Implementation/AWSLambdaUtils.cs
Comment (Kielek) [inline src/OpenTelemetry.Contrib.Instrumentation.AWSLambda/Implementation/AWSLambdaUtils.cs:30]: Can we have here `OpenTelemetry.Instrumentation.AWSLambda`?
Current assembly name is temporary, until 2.0 release.

Comment (samimusallam) [reply src/OpenTelemetry.Contrib.Instrumentation.AWSLambda/Implementation/AWSLambdaUtils.cs:30]: You mean hardcoded, until the release of 2.0?

Comment (Kielek) [reply src/OpenTelemetry.Contrib.Instrumentation.AWSLambda/Implementation/AWSLambdaUtils.cs:30]: Yes, to avoid further change in ActivitySourceName

File: src/OpenTelemetry.Contrib.Instrumentation.AWSLambda/Implementation/AWSLambdaUtils.cs
Comment (Oberon00) [inline src/OpenTelemetry.Contrib.Instrumentation.AWSLambda/Implementation/AWSLambdaUtils.cs:31]: What is the expected result here? OpenTelemetry.Instrumentation.AWSLambda? Why do we remove the .Contrib @Kielek (https://github.com/open-telemetry/opentelemetry-dotnet-contrib/pull/534#discussion_r9298623909)?

Comment (Kielek) [reply src/OpenTelemetry.Contrib.Instrumentation.AWSLambda/Implementation/AWSLambdaUtils.cs:31]: @Oberon00, there is a decision to remove `Contrib` infix for all packages. Only AWS related do not follow this change, but @srprash mentioned that he would like to do this when major version will be increased. See https://github.com/open-telemetry/opentelemetry-dotnet-contrib/issues/196#issuecomment-1092161046

The second thing to consider is task to use common pattern name for all activity source https://github.com/open-telemetry/opentelemetry-dotnet-contrib/issues/484

If we decide to change the activity source for this package, it will be good to make the change only once. So my proposal is to use current package name without `.Contrib`.

**Review by Oberon00 — APPROVED** [review]

File: src/OpenTelemetry.Contrib.Instrumentation.AWSLambda/Implementation/AWSLambdaUtils.cs
Comment (Oberon00) [reply src/OpenTelemetry.Contrib.Instrumentation.AWSLambda/Implementation/AWSLambdaUtils.cs:31]: Since we had a breaking change in #408, I think it would make sense to merge this. Can you please resolve merge conflicts @Kielek?

Comment (Kielek) [reply src/OpenTelemetry.Contrib.Instrumentation.AWSLambda/Implementation/AWSLambdaUtils.cs:31]: @samimusallam is the branch owner. So he should do this. Probably also @open-telemetry/dotnet-contrib-maintainers have possibility to do this.\
If it will be not possible, we should recreate PR from the fork where we can modify branches.

Comment (Oberon00) [reply src/OpenTelemetry.Contrib.Instrumentation.AWSLambda/Implementation/AWSLambdaUtils.cs:31]: Sorry, for some reason I thought you submitted the PR and did not check. So of course I wanted to ask the PR author @samimusallam to resolve the conflicts.

Comment (samimusallam) [reply src/OpenTelemetry.Contrib.Instrumentation.AWSLambda/Implementation/AWSLambdaUtils.cs:34]: @Oberon00 conflicts resolved :)

File: src/OpenTelemetry.Contrib.Instrumentation.AWSLambda/CHANGELOG.md
Comment (Kielek) [inline src/OpenTelemetry.Contrib.Instrumentation.AWSLambda/CHANGELOG.md:7]:
```suggestion
  ([#534](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/pull/534))
```

**Review by Kielek — APPROVED** [review]

**Comment by Oberon00** [comment]
@open-telemetry/dotnet-contrib-maintainers  I'm an "owner" of the component changed here, but not an approver on the repository. What can I do to get this PR merged? I would like to have this merged before requesting an release.

**Comment by reyang** [comment]
> @open-telemetry/dotnet-contrib-maintainers I'm an "owner" of the component changed here, but not an approver on the repository. What can I do to get this PR merged? I would like to have this merged before requesting an release.

The [current process](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/main/CONTRIBUTING.md#how-to-get-prs-merged) requires approval from maintainer/approver. I _guess_ we want it to be changed to "the PR has to receive approval from either the project maintainer/approver, OR the owner of the component (as indicated by the `component_owners.yml` file).

File: src/OpenTelemetry.Contrib.Instrumentation.AWSLambda/CHANGELOG.md
Comment (utpilla) [inline src/OpenTelemetry.Contrib.Instrumentation.AWSLambda/CHANGELOG.md:5]: This is not the assembly name.

Comment (Oberon00) [reply src/OpenTelemetry.Contrib.Instrumentation.AWSLambda/CHANGELOG.md:5]: Right, this should be corrected. But I hope it _is_ consistent with the current guidelines? At least that was the assumption under which I approved it.

**Pull request merged**
